docs/developer/README.md

```diff
@@ -0,0 +1,595 @@
+
+```go=
+type ReferrerStore interface {
+ Name() string
+ ListReferrers(ctx context.Context, subjectReference common.Reference, artifactTypes []string, nextToken string, subjectDesc *ocispecs.SubjectReference) (ListReferrersResult, error)
+ // Used for small objects.
+ GetBlobContent(ctx context.Context, subjectReference common.Reference, digest digest.Digest) ([]byte, error)
+ GetReferenceManifest(ctx context.Context, subjectReference common.Reference, referenceDesc ocispecs.ReferenceDescriptor) (ocispecs.ReferenceManifest, error)
+}
+```
+
+### Requirements
+
+- A referrer store SHOULD be a [plugin](#Plugin-architecture) that implements the interface. The plugin will be integrated in the framework using registration through configuration.
+- Any specific settings required for a store SHOULD be embedded in the plugin configuration that is part of the framework. These settings SHOULD not conflict with reserved properties defined by the framework.
+- There is no ordinal sequencing for the referrers returned by the store.
+- The framework MAY provide a reference implementation for the interface as a builtin plugin. For example, store implementation using OCI registry APIs.
+
+### Open Questions
+
+- If there are no referrers available for a subject, should it return an error or an empty result?
+- Can we assume that the size of the blobs contained within a reference is manageable to download in memory and pass it to the framework? If so, what will be the size limit? Or do we need to support streaming of the content?
+
+### [Referrer Store Plugin and Framework contract specification](https://hackmd.io/wkAGhJgQSCaX4ln9CrXo_w)
+
+## Referrer/Reference Provider
+
+The framework can be configured with multiple referrer stores. A referrers provider provides a capability to query one or more underlying referrer stores to obtain a list of possible refererence artifacts of a particular artifact type. Given an artifact, a set of registered stores will be queried to retrieve all the references. Every reference will be associated with the details of the store it is retrieved from. This  can be used to query further metadata related to a reference like manifest & blobs as part of verifying it.
+
+### Requirements
+
+- Multiple stores MAY be registered through configuration as plugins or internal providers.
+- Provider SHOULD ensure that there is atleast ONE store registered in the configuration.
+- The order of query for references by the  provider MAY be governed by order of registrations of stores in the configuration.
+- Selection of a particular referrers store  is defined below. The same can be used for an artifact to restrict the query to a particular store.
+
+> `MatchingLables` might become the construct that we chose to detemine which stores to query for references.
+
+- Order or selection does not ensure order of return of the references and there will be no ordinal sequencing of references retruned from the provider.
+
+### Sample interface
```

Comment:
General Comment: This document has a lot of the original design proposal for Ratify. Do we need to include this now since we've firmed up on most of the design. Might be confusing for this to be in the developer area. Ideally I think the document should just define what we have now.

Reply:
Agree this whole readme needs to be overhauled, but was out of scope.  Will add note to the document improvements section.